sonar-scala-plugin/pom.xml

```diff
@@ -168,4 +169,80 @@
+                        <artifactId>scala-maven-plugin</artifactId>
+                        <version>3.3.2</version>
+                        <inherited>true</inherited>
+                        <executions>
+                            <execution>
+                                <id>scala-compile-first</id>
+                                <phase>process-resources</phase>
+                                <goals>
+                                    <goal>add-source</goal>
+                                    <goal>compile</goal>
+                                </goals>
+                                <configuration>
+                                    <args>
+                                        <arg>-P:scapegoat:dataDir:${project.build.directory}/scapegoat</arg>
+                                        <arg>-P:scapegoat:reports:scalastyle</arg>
+                                        <arg>-P:scapegoat:overrideLevels:TraversableHead=Warning</arg>
+                                    </args>
+                                    <compilerPlugins>
+                                        <compilerPlugin>
+                                            <groupId>com.sksamuel.scapegoat</groupId>
+                                            <artifactId>scalac-scapegoat-plugin_${scala.binary.version}</artifactId>
+                                            <version>1.3.8</version>
+                                        </compilerPlugin>
+                                    </compilerPlugins>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.scalastyle</groupId>
+                        <artifactId>scalastyle-maven-plugin</artifactId>
+                        <version>1.0.0</version>
+                        <configuration>
+                            <verbose>false</verbose>
+                            <failOnViolation>false</failOnViolation>
+                            <includeTestSourceDirectory>false</includeTestSourceDirectory>
+                            <failOnWarning>false</failOnWarning>
+                            <sourceDirectory>${project.basedir}/src/main/scala</sourceDirectory>
+                            <outputFile>${project.build.directory}/scalastyle-output.xml</outputFile>
+                            <outputEncoding>UTF-8</outputEncoding>
+                        </configuration>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>check</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
```

Comment:
I would have preferred a separate commit for that.

Reply:
You can accept the PR without merging it, and I will split and merge it after the review.